Extend usability of `CMP_OPT_PERMIT_TA_IN_EXTRACERTS_FOR_IR` quirk

DDvO opened the pull request:
As requested in #27888, allow setting the quirk option `CMP_OPT_PERMIT_TA_IN_EXTRACERTS_FOR_IR`,
which had been added for special 3GPP support, also from the `cmp` app.
~~Yet as discussed there, not doing this by default: only if `OPENSSL_CMP_APP_ALLOW_UNSAFE` is defined at compile time.~~

Also make sure that with this quirk option, the new TA(s) are used for whole transaction (also `pkiconf`)
and update and slightly improve its doc.

shahsb commented on crypto/cmp/cmp_vfy.c:
```diff
@@ -358,6 +360,14 @@ static int check_cert_path_3gpp(const OSSL_CMP_CTX *ctx,
+         * on success, add self-issued extraCerts to trust store,
+         * for use with validating any subsequent responses, such as pkiconf
+         */
+        if (valid && (!ossl_cmp_X509_STORE_add1_certs(ctx->trusted, msg->extraCerts,
```
Could you please confirm that the cache is cleared/reset when the transaction ends to avoid cross-transaction contamination?

DDvO replied:
Good point that one should be careful that the exceptional - and insecure - use of extaCerts as trust anchors does not spread to any subsequent transactions!
Changed the implementation of the new extension this way.

shahsb reviewed (commented):
Consider adding a `CMP_CTX_clear_extra_certs_cache(ctx)` API for explicit cleanup (though not required in this PR).

DDvO commented:
> Consider adding a `CMP_CTX_clear_extra_certs_cache(ctx)` API for explicit cleanup (though not required in this PR).

I do not see the need for this. The `extraCertsIn` field in the CMP context will be filled from the `extraCerts` of the first response message in a transaction.
When reusing a CMP context for a further transaction, anyway `OSSL_CMP_CTX_reinit()` needs to be called, which clear it.

openssl-machine commented:
This PR is in a state where it requires action by @openssl/committers but the last update was 30 days ago

openssl-machine commented:
This PR is in a state where it requires action by @openssl/committers but the last update was 61 days ago

DDvO commented on doc/man1/openssl-cmp.pod.in:
```diff
@@ -1529,6 +1547,8 @@ and B<-rsp_crl> options were added in OpenSSL 3.4.
 B<-centralkeygen>, b<-newkeyout>, B<-rsp_key> and
 B<-rsp_keypass> were added in OpenSSL 3.5.
 
+The B<-ta_in_ip_extracerts> quirk option was added in OpenSSL 3.6.
```
updated

DDvO commented:
Rebased on latest `master`, also auto-squashing the fixup commits, which IMO are not of interest for reviewers.

openssl-machine commented:
This PR is in a state where it requires action by @openssl/committers but the last update was 30 days ago

DDvO commented:
Rebased to fix simple doc merge conflict and removed from here the addtion
`if (OSSL_CMP_MSG_get_bodytype(msg) == OSSL_CMP_PKIBODY_IP)` in `cmp_vfy.c`
because this is now covered by #29302.
